verifier/src/qe_report_body.rs

```diff
@@ -0,0 +1,548 @@
+
+        write!(
+            f,
+            "{:pad$}{status} QE Report Body all of the following must be true:",
+            ""
+        )?;
+        let pad = pad + MESSAGE_INDENT;
+        writeln!(f)?;
+        self.mr_signer.fmt_padded(f, pad, &output.value.mr_signer)?;
+        writeln!(f)?;
+        self.isv_prod_id
+            .fmt_padded(f, pad, &output.value.isv_prod_id)?;
+        writeln!(f)?;
+        self.miscellaneous_select
+            .fmt_padded(f, pad, &output.value.miscellaneous_select)?;
+        writeln!(f)?;
+        self.attributes
+            .fmt_padded(f, pad, &output.value.attributes)?;
+        writeln!(f)?;
+        self.isv_svn.fmt_padded(f, pad, &output.value.isv_svn)
+    }
+}
+
+/// Verifier for ensuring a QE(quoting enclave) ISV SVN falls within the
+/// provided [`TcbLevel`]s.
+///
+/// Preforms step 4 of
+/// <https://api.portal.trustedservices.intel.com/documentation#pcs-enclave-identity-v4>
+///
+/// > 4. Determine a TCB status of the Enclave:
+/// >       a. Retrieve a collection of TCB Levels (sorted by ISVSVNs) from
+/// >          tcbLevels field in Enclave Identity structure.
+/// >       b. Go over the list of TCB Levels (descending order) and find the
+/// >          one that has ISVSVN that is lower or equal to the ISVSVN value
+/// >          from SGX Enclave Report.
+/// >       c. If a TCB level is found, read its status from tcbStatus field,
+/// >          otherwise your TCB Level is not supported.
+#[derive(Debug, Clone)]
+struct QeIsvSvnVerifier {
+    tcb_levels: Vec<TcbLevel>,
+}
+
+impl QeIsvSvnVerifier {
+    fn new<'a, I>(tcb_levels: I) -> Self
+    where
+        I: IntoIterator<Item = &'a TcbLevel>,
+    {
+        Self {
+            tcb_levels: tcb_levels.into_iter().cloned().collect(),
+        }
+    }
+}
+
+impl<E: Accessor<IsvSvn>> Verifier<E> for QeIsvSvnVerifier {
+    type Value = (IsvSvn, TcbLevel);
+    fn verify(&self, evidence: &E) -> VerificationOutput<Self::Value> {
+        let isv_svn = evidence.get();
+        let tcb_levels = &self.tcb_levels;
+        let tcb_level = tcb_levels
+            .iter()
+            .find_map(|tcb_level| {
+                if tcb_level.isv_svn().as_ref() <= isv_svn.as_ref() {
+                    Some(tcb_level.clone())
+                } else {
+                    None
+                }
+            })
+            .unwrap_or_default();
+
+        let is_success_bool =
+            tcb_level.status() == AdvisoryStatus::UpToDate && tcb_level.advisory_ids().is_empty();
+        let is_success = if is_success_bool { 1 } else { 0 };
+
+        VerificationOutput::new((isv_svn, tcb_level), is_success.into())
+    }
+}
+
+impl Display for QeIsvSvnVerifier {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "The {} should correspond to an `UpToDate` level with no advisories, from: {:?}",
+            IsvSvn::spaced_struct_name(),
+            self.tcb_levels
```

Comment:
I'm not sure if/how much we want to invest in making a Display for tcb_levels. 
I also think this output can get a bit long on the line, but really if someone is going to display the verification result it's going to be pretty big overall.

Reply:
I think we can leave it like this and if this proves to be too long then provide a nicer Display for it